src/rng.rs

```diff
@@ -393,14 +393,20 @@ impl Fill for [u8] {
     }
 }
 
-macro_rules! impl_fill {
+// This macro is unsafe to call: target types must support transmute from
+// random bits (i.e. all bit representations are valid).
+macro_rules! unsafe_impl_fill {
     () => {};
     ($t:ty) => {
         impl Fill for [$t] {
-            #[inline(never)] // in micro benchmarks, this improves performance
             fn fill<R: Rng + ?Sized>(&mut self, rng: &mut R) {
                 if self.len() > 0 {
-                    rng.fill_bytes(self.as_mut_bytes());
+                    rng.fill_bytes(unsafe {
```

Comment:
Safety comment? You'll need to justify that you've upheld [this safety contract](https://doc.rust-lang.org/std/slice/fn.from_raw_parts_mut.html#safety). You may find the following to be a helpful rephrasing of similar requirements: https://doc.rust-lang.org/1.82.0/std/ptr/index.html#pointer-to-reference-conversion

I'd recommend doing [`#![deny(clippy::undocumented_unsafe_blocks)]`](https://rust-lang.github.io/rust-clippy/rust-1.85.0/index.html#undocumented_unsafe_blocks) for the whole crate.

Reply:
I added a short safety comment and denied the clippy lint you suggested for `rand` and `rand_core`.